src/symbols-manual.js

```diff
@@ -238,6 +238,55 @@ Rocky.addManualSymbols = function(obj) {
     });
   };
 
+  obj.gpath_create = function(config) {
+    // handle case where config is an array
+    if (!('points' in config)) {
+      config = {points: config};
+    }
+    config.points = config.points.map(function(pt) {return obj.GPoint(pt);});
+    config.offset = obj.GPoint(config.offset || [0, 0]);
+    config.rotation = config.rotation || 0;
+    config.captureCPointer = function() {
+      // we're describing a GPath structure followed by the actual points
+      // typedef struct GPath {
+      //   uint32_t num_points; // 0
+      //   GPoint *points;      // 4
+      //   int32_t rotation;    // 8
+      //   GPoint offset;       // 12
+      // } GPath;               // 16+ actual points
+      var needed_bytes = 16 + this.points.length * 4;
+      var ptr = obj.module._malloc(needed_bytes);
+      obj.module.setValue(ptr + 0, this.points.length, 'i32');
```

Comment:
IMO, it's ok if the JS part of the binding has knowledge about the in-memory representation. The best I can think of to offload this to C would be a builder, e.g.
```
// special EMXGPath type to express that the otherwise `const *points` array is writable
// and follows directly after the GPath
EMXGPath *emx_gpath_create(uint32_t max_points, int32_t rotation, int16_t offset_x, int16_t offset_y);
bool emx_gpath_add_point(EMXGPath *path, int16_t pt_x, int16 py_y);
```
This wouldn't make the JS part much shorter, though. I prefer the logic to sit on the JS side and would rather invest into a more convenient way for JS to interpret memory regions as C structs.